mobile/src/people/departments/departments-horizontal-scrollable-list.tsx

```diff
@@ -1,135 +1,118 @@
-import { Animated, View, ScrollView, Dimensions, NativeScrollEvent, NativeSyntheticEvent } from 'react-native';
-import { EmployeeCardWithAvatar } from '../employee-card-with-avatar';
+import { View, ScrollView, Dimensions, NativeScrollEvent, NativeSyntheticEvent } from 'react-native';
+import { EmployeeCardWithAvatar } from './employee-card-with-avatar';
 import { Employee } from '../../reducers/organization/employee.model';
 import { EmployeesStore } from '../../reducers/organization/employees.reducer';
 import { Department } from '../../reducers/organization/department.model';
-import { employeesAZComparer } from '../employee-comparer';
+import { Set } from 'immutable';
 
 interface DepartmentsHScrollableListProps {
-    treeLevel?: number;
-    headDepartmentId?: string;
-    headDepartmentChiefId?: string;
-    departments?: Department[];
-    departmentsLists?: DepartmentsListStateDescriptor;
-    focusOnDepartmentWithId?: string;
-    currentFocusedDepartmentId?: string;
-    employees?: EmployeesStore;
-    topOffset?: number;
-    requestEmployeesForDepartment: (departmentId: string) => void;
-    updateDepartmentsBranch: (departmentId: string, focusOnEmployeesLust?: boolean) => void;
+    headDepartment: Department;
+    departments: Department[];
+    departmentsLists: DepartmentsListStateDescriptor;
+    employees: EmployeesStore;
+    updateDepartmentsBranch: (departmentId: string) => void;
     onItemClicked: (e: Employee) => void;
-    employeesPredicate: (employee: Employee) => boolean;
+    key: string;
 }
 
 interface ScrollViewComponent extends Component {
-    scrollTo(y?: number | { x?: number; y?: number; animated?: boolean }, x?: number, animated?: boolean): void;
-    scrollToEnd(): void;
+    scrollTo(to : {y?: number, x?: number, animated?: boolean}): void;
 }
 
 export interface DepartmentsListStateDescriptor {
     currentPage: number;
-    // forceComponentRender: boolean;
-    // manualScrollMode: boolean;
 }
 
 export class DepartmentsHScrollableList extends Component<DepartmentsHScrollableListProps> {
-    private scrollView: ScrollViewComponent;
     private employeeCards: EmployeeCardWithAvatar[];
-    private animatedValue: Animated.Value;
-
-    public shouldComponentUpdate(nextProps: DepartmentsHScrollableListProps) {	
-        if (this.props.departmentsLists !== nextProps.departmentsLists) {	
-            return true;	
-        } else {	
-            const employees = this.props.employees.employeesById.filter(this.props.employeesPredicate);	
-            const nextEmployees = nextProps.employees.employeesById.filter(nextProps.employeesPredicate);	
+    private scrollView: ScrollViewComponent;
 
-            return !employees.equals(nextEmployees);
-        }	
+    public componentDidMount() {
+        this.setScroll();
+    }
+    public componentDidUpdate(prevProps: DepartmentsHScrollableListProps) {
+        if (!Set(prevProps.departments).equals(Set(this.props.departments)) ||
```

Comment:
I guess Set(prevProps.departments) and Set(this.props.departments) are always not equal. It will work as you're expecting if departments are immutablejs-like objects